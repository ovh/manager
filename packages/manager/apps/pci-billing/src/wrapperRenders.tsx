import { QueryClient, QueryClientProvider } from '@tanstack/react-query';
import {
  ShellContext,
  ShellContextType,
} from '@ovh-ux/manager-react-shell-client';
import { vi } from 'vitest';

export const shellContext = {
  environment: {
    getUser: () => ({ ovhSubsidiary: 'FR', currency: 'â‚¬' }),
  },
  shell: {
    navigation: {
      getURL: vi.fn(),
    },
  },
};

const queryClient = new QueryClient({
  defaultOptions: {
    mutations: {
      retry: false,
    },
    queries: {
      retry: false,
    },
  },
});

export const wrapper = ({ children }) => (
  <QueryClientProvider client={queryClient}>
    <ShellContext.Provider
      value={(shellContext as unknown) as ShellContextType}
    >
      {children}
    </ShellContext.Provider>
  </QueryClientProvider>
);
