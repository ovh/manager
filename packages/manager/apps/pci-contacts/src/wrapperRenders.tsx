import {
  ShellContext,
  ShellContextType,
} from '@ovh-ux/manager-react-shell-client';
import { QueryClient, QueryClientProvider } from '@tanstack/react-query';
import { ReactNode } from 'react';
import { MemoryRouter } from 'react-router-dom';
import { vi } from 'vitest';

export const shellContext = ({
  environment: {
    getUser: () => ({ ovhSubsidiary: 'FR' }),
    getRegion: () => 'EU',
  },
  shell: {
    navigation: {
      navigateTo: vi.fn(),
      getURL: vi.fn().mockImplementation((app, path) => {
        if (app === 'dedicated' && path === '#/billing/history') {
          return Promise.resolve('https://billing-url');
        }
        return Promise.resolve('');
      }),
    },
  },
} as unknown) as ShellContextType;

export const createWrapper = (
  contextValue: ShellContextType = shellContext,
) => {
  const queryClient = new QueryClient({
    defaultOptions: {
      queries: {
        staleTime: 300_000,
      },
    },
  });

  const Wrapper = ({ children }: { children: ReactNode }) => (
    <QueryClientProvider client={queryClient}>
      <ShellContext.Provider value={contextValue}>
        <MemoryRouter>{children}</MemoryRouter>
      </ShellContext.Provider>
    </QueryClientProvider>
  );
  return Wrapper;
};
