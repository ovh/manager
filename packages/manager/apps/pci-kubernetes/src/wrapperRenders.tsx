import { QueryClient, QueryClientProvider } from '@tanstack/react-query';
import {
  ShellContext,
  ShellContextType,
} from '@ovh-ux/manager-react-shell-client';
import { vi } from 'vitest';

export const shellContext = {
  environment: {
    getRegion: vi.fn().mockReturnValue('EU'),
    getUser: () => ({
      ovhSubsidiary: 'mocked_ovhSubsidiary',
      currency: { code: 'EUR' },
    }),
  },
  shell: {
    navigation: {
      getURL: vi.fn(),
    },
  },
};

const queryClient = new QueryClient({
  defaultOptions: {
    queries: {
      retry: false,
    },
  },
});

export const wrapper = ({ children }) => (
  <QueryClientProvider client={queryClient}>
    <ShellContext.Provider
      value={(shellContext as unknown) as ShellContextType}
    >
      {children}
    </ShellContext.Provider>
  </QueryClientProvider>
);
