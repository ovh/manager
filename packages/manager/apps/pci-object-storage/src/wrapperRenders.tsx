import { QueryClient, QueryClientProvider } from '@tanstack/react-query';
import {
  ShellContext,
  ShellContextType,
} from '@ovh-ux/manager-react-shell-client';
import { vi } from 'vitest';
import { createMemoryRouter, RouterProvider } from 'react-router-dom';
import React from 'react';

export const shellContext = {
  environment: {
    getUser: () => ({ ovhSubsidiary: 'mocked_ovhSubsidiary' }),
  },
  shell: {
    navigation: {
      getURL: vi.fn(),
    },
    tracking: {
      trackPage: vi.fn(),
      trackClick: vi.fn(),
    },
  },
};

const queryClient = new QueryClient({
  defaultOptions: {
    mutations: {
      retry: false,
    },
    queries: {
      retry: false,
    },
  },
});

export const wrapper = ({ children }) => (
  <QueryClientProvider client={queryClient}>
    <ShellContext.Provider
      value={(shellContext as unknown) as ShellContextType}
    >
      {children}
    </ShellContext.Provider>
  </QueryClientProvider>
);

export const mockShellContext = {
  environment: {
    getRegion: vi.fn().mockReturnValue('EU'),
    getUser: vi.fn().mockReturnValue({ ovhSubsidiary: 'mocked_ovhSubsidiary' }),
    getApplication: vi.fn(),
  },
  shell: {
    environment: {
      getRegion: vi.fn().mockReturnValue('EU'),
      getUser: vi.fn().mockReturnValue({}),
      getApplication: vi.fn(),
    },
    navigation: {
      getURL: vi.fn(),
    },
    tracking: {
      trackPage: vi.fn(),
      trackClick: vi.fn(),
    },
  },
};

export const wrapperShow = ({ children }: { children: React.ReactNode }) => {
  const router = createMemoryRouter(
    [
      {
        path: '/',
        element: children,
        handle: {
          crumb: () => 'Home',
        },
      },
    ],
    {
      initialEntries: ['/'],
      future: {
        v7_startTransition: true,
        v7_relativeSplatPath: true,
      } as any,
    },
  );

  return (
    <QueryClientProvider client={queryClient}>
      <ShellContext.Provider value={mockShellContext as any}>
        <RouterProvider router={router} />
      </ShellContext.Provider>
    </QueryClientProvider>
  );
};

export const wrapperOffsiteReplication = ({
  children,
}: {
  children: React.ReactNode;
}) => (
  <QueryClientProvider client={queryClient}>
    <ShellContext.Provider value={mockShellContext as any}>
      {children}
    </ShellContext.Provider>
  </QueryClientProvider>
);
