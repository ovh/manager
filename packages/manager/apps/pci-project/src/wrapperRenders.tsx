import {
  ShellContext,
  ShellContextType,
} from '@ovh-ux/manager-react-shell-client';
import { QueryClient, QueryClientProvider } from '@tanstack/react-query';
import { ReactNode } from 'react';
import { MemoryRouter } from 'react-router-dom';
import { vi } from 'vitest';

export const shellContext = ({
  environment: {
    getUser: () => ({ ovhSubsidiary: 'FR' }),
    getRegion: () => 'EU',
  },
  shell: {
    navigation: {
      navigateTo: vi.fn(),
      getURL: vi.fn().mockImplementation((app, path) => {
        if (app === 'dedicated' && path === '#/billing/history') {
          return Promise.resolve('https://billing-url');
        }
        if (app === 'dedicated' && path === '#/contacts/services') {
          return Promise.resolve('https://contacts-url');
        }
        return Promise.resolve('https://mocked-url');
      }),
    },
  },
} as unknown) as ShellContextType;

export const createWrapper = (
  contextValue: ShellContextType = shellContext,
) => {
  const queryClient = new QueryClient({
    defaultOptions: {
      queries: {
        staleTime: 300_000,
      },
    },
  });

  const Wrapper = ({ children }: { children: ReactNode }) => (
    <QueryClientProvider client={queryClient}>
      <ShellContext.Provider value={contextValue}>
        <MemoryRouter>{children}</MemoryRouter>
      </ShellContext.Provider>
    </QueryClientProvider>
  );
  return Wrapper;
};
