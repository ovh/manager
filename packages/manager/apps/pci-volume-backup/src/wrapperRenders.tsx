import { ReactNode } from 'react';
import { QueryClientProvider } from '@tanstack/react-query';
import {
  ShellContext,
  ShellContextType,
} from '@ovh-ux/manager-react-shell-client';
import { vi } from 'vitest';
import { ErrorBoundary } from 'react-error-boundary';
import queryClient from './queryClient';

export const shellContext = ({
  environment: {
    getUser: () => ({ ovhSubsidiary: 'mocked_ovhSubsidiary' }),
  },
  shell: {
    navigation: {
      navigateTo: vi.fn(),
      getURL: vi.fn(),
    },
  },
} as unknown) as ShellContextType;

export const createWrapper = () => {
  const Wrapper = ({ children }: { children: ReactNode }) => (
    <QueryClientProvider client={queryClient}>
      <ShellContext.Provider value={shellContext}>
        <ErrorBoundary fallbackRender={() => <div>error boundary</div>}>
          {children}
        </ErrorBoundary>
      </ShellContext.Provider>
    </QueryClientProvider>
  );
  return Wrapper;
};
