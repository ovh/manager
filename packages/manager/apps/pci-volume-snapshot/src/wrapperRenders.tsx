import { ReactNode } from 'react';
import { QueryClient, QueryClientProvider } from '@tanstack/react-query';
import {
  ShellContext,
  ShellContextType,
} from '@ovh-ux/manager-react-shell-client';
import { vi } from 'vitest';
import { ErrorBoundary } from 'react-error-boundary';

export const shellContext = {
  environment: {
    getUser: () => ({ ovhSubsidiary: 'mocked_ovhSubsidiary' }),
  },
  shell: {
    navigation: {
      getURL: vi.fn(),
    },
  },
};

export const createWrapper = () => {
  const queryClient = new QueryClient({
    defaultOptions: {
      mutations: {
        retry: false,
      },
      queries: {
        retry: false,
      },
    },
  });
  const Wrapper = ({ children }: { children: ReactNode }) => (
    <QueryClientProvider client={queryClient}>
      <ShellContext.Provider
        value={(shellContext as unknown) as ShellContextType}
      >
        <ErrorBoundary fallbackRender={() => <div>error boundary</div>}>
          {children}
        </ErrorBoundary>
      </ShellContext.Provider>
    </QueryClientProvider>
  );
  return Wrapper;
};
