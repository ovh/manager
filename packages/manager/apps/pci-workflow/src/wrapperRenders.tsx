import { QueryClient, QueryClientProvider } from '@tanstack/react-query';

const queryClient = new QueryClient();
export const wrapper = ({ children }) => (
  <QueryClientProvider client={queryClient}>{children}</QueryClientProvider>
);
