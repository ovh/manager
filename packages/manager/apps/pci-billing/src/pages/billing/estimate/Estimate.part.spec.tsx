import { describe, vi } from 'vitest';
import React from 'react';
import { render } from '@testing-library/react';
import {
  EstimatePart,
  TEstimateProps,
} from '@/pages/billing/estimate/Estimate.part';
import { wrapper } from '@/wrapperRenders';

vi.mock('@ovhcloud/ods-components/react', async () => {
  const {
    OsdsSpinner: ActualOsdsSpinner,
    OsdsText: ActualOsdsText,
    ...rest
  } = await vi.importActual('@ovhcloud/ods-components/react');
  const [OsdsSpinnerElem, OsdsTextElem] = [
    ActualOsdsSpinner as React.ElementType,
    ActualOsdsText as React.ElementType,
  ];
  return {
    ...rest,
    OsdsSpinner: vi
      .fn()
      .mockImplementation(({ ...props }) => (
        <OsdsSpinnerElem {...props} data-testid="osds-spinner" />
      )),
    OsdsText: vi
      .fn()
      .mockImplementation(({ ...props }) => (
        <OsdsTextElem {...props} data-testid="osds-text" />
      )),
  };
});

vi.mock('react-i18next', async () => {
  const { ...rest } = await vi.importActual('react-i18next');
  return {
    ...rest,
    useTranslation: vi.fn().mockImplementation((namespace: string) => ({
      t: vi.fn().mockImplementation((key: string) => `${namespace} | ${key}`),
    })),
  };
});

const PROPS: TEstimateProps = {
  currency: {
    symbol: '€',
    code: 'EUR',
    format: '€0,0.00',
  },
  totalHourlyPrice: 75,
  totalMonthlyPrice: 690,
  totalPrice: 75 + 690,
  isPending: false,
  locale: 'enGB',
};

const renderComponent = (props: TEstimateProps = PROPS) =>
  render(<EstimatePart {...props} />, { wrapper });

describe('EstimatePart', () => {
  describe('Prices rae not pending', () => {
    it('should render the component if locale is found', () => {
      const { container } = renderComponent();
      expect(container).toMatchSnapshot();
    });

    it('should render the component if locale is not found', () => {
      const { container } = renderComponent({
        ...PROPS,
        locale: 'notFound',
      });
      expect(container).toMatchSnapshot();
    });
  });

  it('should render the component if is pending', () => {
    const { container } = renderComponent({
      ...PROPS,
      isPending: true,
    });
    expect(container).toMatchSnapshot();
  });
});
