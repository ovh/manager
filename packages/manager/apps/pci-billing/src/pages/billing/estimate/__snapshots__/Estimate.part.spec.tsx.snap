// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`EstimatePart > Prices rae not pending > should render the component if locale is found 1`] = `
<div>
  <div>
    <osds-text
      color="text"
      data-testid="osds-text"
      level="heading"
      size="500"
    >
      consumption | cpbc_tab_forecast
    </osds-text>
    <p>
      <osds-text
        color="text"
        data-testid="osds-text"
        level="body"
        size="400"
      >
        estimate | cpbe_estimate_summary_header
         
        <strong>
          765.00
           
          €
        </strong>
      </osds-text>
    </p>
    <div>
      <p>
        <osds-text
          color="text"
          data-testid="osds-text"
          level="body"
          size="400"
        >
          <span>
            estimate | cpbe_estimate_summary_renew_monthly_sub_label
          </span>
        </osds-text>
      </p>
      <p>
        <osds-text
          color="text"
          data-testid="osds-text"
          level="body"
          size="400"
        >
          <span>
            estimate | cpbe_estimate_summary_hourly_consumption_label
          </span>
        </osds-text>
      </p>
    </div>
  </div>
</div>
`;

exports[`EstimatePart > Prices rae not pending > should render the component if locale is not found 1`] = `
<div>
  <div>
    <osds-text
      color="text"
      data-testid="osds-text"
      level="heading"
      size="500"
    >
      consumption | cpbc_tab_forecast
    </osds-text>
    <p>
      <osds-text
        color="text"
        data-testid="osds-text"
        level="body"
        size="400"
      >
        estimate | cpbe_estimate_summary_header
         
        <strong>
          765.00
           
          €
        </strong>
      </osds-text>
    </p>
    <div>
      <p>
        <osds-text
          color="text"
          data-testid="osds-text"
          level="body"
          size="400"
        >
          <span>
            estimate | cpbe_estimate_summary_renew_monthly_sub_label
          </span>
        </osds-text>
      </p>
      <p>
        <osds-text
          color="text"
          data-testid="osds-text"
          level="body"
          size="400"
        >
          <span>
            estimate | cpbe_estimate_summary_hourly_consumption_label
          </span>
        </osds-text>
      </p>
    </div>
  </div>
</div>
`;

exports[`EstimatePart > should render the component if is pending 1`] = `
<div>
  <div>
    <osds-text
      color="text"
      data-testid="osds-text"
      level="heading"
      size="500"
    >
      consumption | cpbc_tab_forecast
    </osds-text>
    <p>
      <osds-text
        color="text"
        data-testid="osds-text"
        level="body"
        size="400"
      >
        estimate | cpbe_estimate_summary_header
         
        <strong>
          765.00
           
          €
        </strong>
      </osds-text>
    </p>
    <osds-spinner
      data-testid="osds-spinner"
      inline="true"
    />
  </div>
</div>
`;
